File: robocode2-bot-api-java/src/main/java/net/robocode2/events/BotHitWallEvent.java
package net.robocode2.events;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Value;

/** Event occurring when the bot has hit a wall */
@Value
@EqualsAndHashCode(callSuper = true)
@Builder
public class BotHitWallEvent extends GameEvent {
    /** ID of the victim bot that hit the wall */
    int victimId;
}

File: robocode-tankroyale-bot-api-csharp/src/events/BotHitWallEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi
{
  /// <summary>
  /// Event occurring when the bot has hit a wall.
  /// </summary>
  public sealed class BotHitWallEvent : Event
  {
    /// <summary>
    /// Initializes a new instance of the BotHitWallEvent class.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    [JsonConstructor]
    public BotHitWallEvent(int turnNumber) : base(turnNumber) { }
  }
}